.github/workflows/update-dependencies.yaml

```diff
@@ -0,0 +1,115 @@
+name: update-dependencies
+on:
+  workflow_call:
+    inputs:
+      component:
+        type: string
+        required: true
+      write-lock:
+        type: string
+        required: true
+      label:
+        type: string
+        required: true
+
+jobs:
+  init:
+    runs-on: [ "self-hosted", "builder" ]
+    outputs:
+      metadata: ${{ steps.init.outputs.metadata }}
+      parameters: ${{ steps.init.outputs.parameters }}
+    steps:
+      - id: init
+        name: Init Workflow
+        uses: hawk-ai-aml/github-actions/workflow-init@master
+        with:
+          component: ${{ inputs.component }}
+          profile: hawk
+          update-kustomize: false
+          
+  lock-dependencies:
+    needs: [ init ]
+    runs-on: [ "self-hosted", "builder" ]
+    outputs:
+      result: ${{ steps.commit-and-push.outputs.result }}
+    steps:
+      - id: init
+        name: Init Job
+        uses: hawk-ai-aml/github-actions/job-init@master
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ fromJson(needs.init.outputs.parameters).java-version }}
+          distribution: temurin
+
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.REPO_ACCESS_PAT }}
+          ref: update-dependencies
+
+      - name: Fetch and reset update-dependencies to master
+        run: |
+          git fetch origin master
+          git reset --hard origin/master
+
+      - name: Perform dependency resolution and write new lockfiles
+        run: |
+          ${{ inputs.write-lock }}
+          git status
+      - name: Commit, push, create pull request
+        id: commit-and-push
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "github-actions@hawk.ai"
+          git add .
+          
+          if ! git diff --quiet HEAD; then
+            git commit -m "build(gradle): Update gradle.lockfile"
+            git push --force origin update-dependencies
+            echo "result=success" >> $GITHUB_OUTPUT
+          else 
+            echo "result=skipped" >> $GITHUB_OUTPUT
+            echo no dynamic dependencies updates
+          fi
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  create-pr:
+    runs-on: [ "self-hosted", "builder" ]
+    needs: [ init , lock-dependencies ]
+    if: needs.lock-dependencies.outputs.result == 'success'
+    steps:
+      - name: Create pull request
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const { repo, owner } = context.repo;
+            const pr_list = await github.rest.pulls.list({
+              owner,
+              repo,
+              head: 'hawk-ai-aml:update-dependencies',
+              base: 'master'
+            });
+
+            console.log(pr_list)
+
+            if (pr_list.data.length == 0) {
```

Comment:
So we only create the PR when there is no update-dependencies PR exists? What happens when there is a PR not merged? We need to first merge and trigger this job manually to get the latest right?

Reply:
> So we only create the PR when there is no update-dependencies PR exists? 

Correct

> What happens when there is a PR not merged? We need to first merge and trigger this job manually to get the latest right?

Then we do not create a PR, nevertheless, PR we'll contain the newest changes, because in the previous job we push to update-dependencies the newest changes from the lockfiles